release/nightly_tests/dataset/benchmark.py

```diff
@@ -1,182 +1,79 @@
-import gc
 import json
 import os
 import time
-from typing import Callable, Dict
-
-from ray.data.dataset import Dataset
-from typing import Any
-
+from typing import Callable, Dict, Any, Optional
 
 from enum import Enum
 
-import pyarrow as pa
-import pandas as pd
-
 
 class BenchmarkMetric(Enum):
     RUNTIME = "time"
     NUM_ROWS = "num_rows"
     THROUGHPUT = "tput"
     ACCURACY = "accuracy"
 
-    # Extra metrics not matching the above categories/keys, stored as a Dict[str, Any].
-    EXTRA_METRICS = "extra_metrics"
 
+def run_benchmark(
+    benchmark_fn: Callable[[], Optional[Dict]], config: Optional[Dict[str, Any]] = None
+):
+    """Run a benchmark and write the results to the appropriate output file.
+
+    By default, this function measures the total runtime of the benchmark. You can
+    specify additional metrics by returning a dictionary from the benchmark function.
+
+    Example:
+
+        .. testcode::
+
+            import time
+            from benchmark import run_benchmark
+
+            sleep_s = 1
 
-class Benchmark:
-    """Utility class used for Ray Data release tests and benchmarks, which works
-    for both local and distributed benchmarking. When run on the nightly release
-    test pipeline, the results are written to our internal database, which
-    can then be rendered on the dashboard. Usage tips:
+            def benchmark_sleep():
+                time.sleep(sleep_s)
+                return {"extra_metric": 42}
 
-    A typical workflow would be:
+            run_benchmark(benchmark_sleep, config={"sleep_s": sleep_s})
```

Comment:
The meaning of `config` was unintuitive to me. I had to read to code to understand that the config is used to write benchmark results. 
Maybe we can force the benchmark fn to pass the configs as kwargs. 
That way we can also avoid potential inconsistency between the logged config and the actual config.